KomaMRICore/src/simulation/Functors.jl

```diff
@@ -99,19 +97,23 @@ See also [`f32`](@ref).
 """
 f64(m) = paramtype(Float64, m)
 
+# Koma motion-related adapts
+adapt_storage(backend::KA.GPU, xs::MotionList) = MotionList(gpu.(xs.motions, Ref(backend)))
+adapt_storage(T::Type{<:Real}, xs::NoMotion) = NoMotion{T}()
+adapt_storage(T::Type{<:Real}, xs::MotionList) = MotionList(paramtype.(T, xs.motions))
+adapt_storage(T::Type{<:Real}, xs::Motion) = Motion(paramtype(T, xs.action), paramtype(T, xs.time), xs.spins)
```

Comment:
Is this necessary? What happened with the ranges/vector?

Reply:
The first `adapt` to GPU is indeed necessary. If I don't use it, the function |> gpu will try to convert the MotionList.motions attribute (which is a `Vector{Motion}`) to a `GPUArray{Motion}`, and we don't want that. That is also why I am wrapping the motion array inside the `MotionList` struct.
I will check for the other three adapts.

Reply:
What about using `gpu` instead of `adapt`?

Reply:
The `adapt` function does not make use of the `fmap` function, so it would not adapt recursively.
Concerning the necessity of the MotionList, I also think that there must be a way to omit it, and just let everything as it was initially stated in https://github.com/JuliaHealth/KomaMRI.jl/issues/410.
The main problem of this was that doing `obj.motion |> gpu` tried to convert that `Vector{Motion}`) to a `GPUArray{Motion}`, and if I defined a new adapt method like: 
`adapt_storage(backend::KA.GPU, xs::Vector{Motion}) = gpu.(backend, xs)`
it produced a conflict with other defined methods. We can see this later, as it maybe has an easy solution that I am not being able to see.

Reply:
As it is now, I can only omit the 4th line: the adapt to `Motion`

Reply:
Ok, should I wait for the commit?

Reply:
e416800d1165086a399412f8c4ea41b58099f1d3

Reply:
Oh ok, you did both in one commit, I just saw the AbstractMotion part.